packages/block-library/src/image/block.json

```diff
@@ -9,6 +9,10 @@
 	"keywords": [ "img", "photo", "picture" ],
 	"textdomain": "default",
 	"attributes": {
+		"temporaryUrl": {
+			"type": "string",
+			"isTemporary": true
```

Comment:
Is this the right way to flag this kind of attributes?

Reply:
is there a way to represent this more like "don't persist"? Historically we have tried to use state for attributes that are ephemeral. Representing these ephemeral attributes at the block object level seems slightly wrong.

Reply:
> is there a way to represent this more like "don't persist"?

This is the way to represent "don't persist". 

> Historically we have tried to use state for attributes that are ephemeral. Representing these ephemeral attributes at the block object level seems slightly wrong.

The problem here is that it's not possible to use local state, and that's why the issue persisted for a long time. When you drag and drop a file, you want to create the block immediately (immediate feedback) and creating blocks means setting attributes.

I went back and forth between representing this as an attribute that has a special behavior (don't persist) and a new property on the block object `localState` or something that is basically just like attributes but treated separately.

Ultimately, I think making it an attribute is better because:

 - The impact of introducing a new property on the block object is way too big and the benefits are small
 - We may have other kind of attributes in the future that have different behaviors (for instance attributes that impact undo/redo in a certain way...). In the past we discussed with @noisysocks and @gziolo several kind of other attributes, maybe they recall.

Reply:
You've probably already considered it, but another option is to still use the same `url` attribute and have a way to mark it as not persisting. Maybe even a different version of `setAttributes` function.

I guess ultimately that may still end up being more complicated (both for us and block implementers) than a separate attribute and might not solve the other use cases you mentioned, though it could also work alongside the idea of permanently temporary attributes.

Reply:
> and have a way to mark it as not persisting.

Yeah, this was my thinking. Also agreed it might be more complicated than a separate attribute, but it seems less elegant overall.

What about `isLocal` instead of `isTemporary`?